nodes/explorer/src/api_backend/store.rs

```diff
@@ -44,3 +47,163 @@ where
+        Ok(from) => match from {
+            Some(from) => from,
+            None => {
+                return IntoResponse::into_response((StatusCode::NOT_FOUND, "from block not found"))
+            }
+        },
+        Err(e) => {
+            return IntoResponse::into_response((StatusCode::INTERNAL_SERVER_ERROR, e.to_string()))
+        }
+    };
+
+    // check if to is valid
+    match to {
+        Either::Left(to) => match storage::block_req::<S, Tx>(&store, to).await {
+            Ok(to) => match to {
+                Some(to) => handle_to::<S, Tx>(store, from, to).await,
+                None => IntoResponse::into_response((StatusCode::NOT_FOUND, "to block not found")),
+            },
+            Err(e) => {
+                IntoResponse::into_response((StatusCode::INTERNAL_SERVER_ERROR, e.to_string()))
+            }
+        },
+        Either::Right(depth) => handle_depth::<S, Tx>(store, from, depth).await,
+    }
+}
+
+async fn handle_depth<S, Tx>(
+    store: OverwatchHandle,
+    from: Block<Tx, Certificate>,
+    depth: usize,
+) -> Response
+where
+    Tx: Transaction
+        + Clone
+        + Debug
+        + Eq
+        + Hash
+        + Serialize
+        + DeserializeOwned
+        + Send
+        + Sync
+        + 'static,
+    <Tx as Transaction>::Hash: std::cmp::Ord + Debug + Send + Sync + 'static,
+    S: StorageSerde + Send + Sync + 'static,
+{
+    let mut cur = Some(from.header().parent());
+    let mut blocks = Vec::new();
+    while blocks.len() < depth {
+        if let Some(id) = cur {
+            let block = match storage::block_req::<S, Tx>(&store, id).await {
+                Ok(block) => block,
+                Err(e) => {
+                    return IntoResponse::into_response((
+                        StatusCode::INTERNAL_SERVER_ERROR,
+                        e.to_string(),
+                    ))
+                }
+            };
+
+            match block {
+                Some(block) => {
+                    cur = Some(block.header().parent());
+                    blocks.push(block);
+                }
+                None => {
+                    cur = None;
+                }
+            }
+        } else {
+            break;
+        }
+    }
+
+    IntoResponse::into_response((StatusCode::OK, ::axum::Json(blocks)))
+}
+
+async fn handle_to<S, Tx>(
+    store: OverwatchHandle,
+    from: Block<Tx, Certificate>,
+    to: Block<Tx, Certificate>,
+) -> Response
+where
+    Tx: Transaction
+        + Clone
+        + Debug
+        + Eq
+        + Hash
+        + Serialize
+        + DeserializeOwned
+        + Send
+        + Sync
+        + 'static,
+    <Tx as Transaction>::Hash: std::cmp::Ord + Debug + Send + Sync + 'static,
+    S: StorageSerde + Send + Sync + 'static,
+{
+    let mut cur = Some(from.header().parent());
+    let mut blocks = Vec::new();
+    while let Some(id) = cur {
+        if id == to.header().id {
+            break;
+        }
```

Comment:
~Maybe we should check first if `to` is actually a valid id?~ ok, I see we check before the function call.